## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 444428`
Commit: [`1b675189448bc489c96182a7a597c14d0c37265a`](https://github.com/NixOS/nixpkgs/commit/1b675189448bc489c96182a7a597c14d0c37265a) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/1b675189448bc489c96182a7a597c14d0c37265a..pull/444428/head))
Merge: [`41d04e7e9b7a5bcb4c5ba265c1c36900c0650dfa`](https://github.com/NixOS/nixpkgs/commit/41d04e7e9b7a5bcb4c5ba265c1c36900c0650dfa)

Logs: https://github.com/liberodark/nixpkgs-review-gha/actions/runs/17869512367


---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>linuxPackages_lqx.bcachefs (linuxKernel.packages.linux_lqx.bcachefs)</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>bcachefs-tools</li>
    <li>bcachefs-tools.dkms</li>
    <li>linuxPackages_latest.bcachefs (linuxKernel.packages.linux_6_16.bcachefs)</li>
    <li>linuxPackages_latest-libre.bcachefs (linuxKernel.packages.linux_latest_libre.bcachefs)</li>
    <li>linuxPackages_xanmod_latest.bcachefs (linuxKernel.packages.linux_xanmod_latest.bcachefs, linuxPackages_xanmod_stable.bcachefs)</li>
    <li>linuxPackages_zen.bcachefs (linuxKernel.packages.linux_zen.bcachefs)</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>linuxPackages_lqx.bcachefs</summary>
<pre>  CC [M]  src/fs/bcachefs/io_read.o
  CC [M]  src/fs/bcachefs/io_misc.o
  CC [M]  src/fs/bcachefs/io_write.o
  CC [M]  src/fs/bcachefs/journal.o
src/fs/bcachefs/io_write.c: In function &#x27;__wp_update_state&#x27;:
src/fs/bcachefs/io_write.c:653:32: error: &#x27;struct task_struct&#x27; has no member named &#x27;se&#x27;
  653 |                 u64 runtime = p-&gt;se.sum_exec_runtime +
      |                                ^~
src/fs/bcachefs/io_write.c:654:33: error: &#x27;struct task_struct&#x27; has no member named &#x27;se&#x27;
  654 |                         (now - p-&gt;se.exec_start);
      |                                 ^~
make[5]: *** [/nix/store/h9xc6fvrqbp3n2wy8m7g4pzyv2r5zz5m-linux-lqx-6.16.7-dev/lib/modules/6.16.7-lqx1/source/scripts/Makefile.build:287: src/fs/bcachefs/io_write.o] Error 1
make[5]: *** Waiting for unfinished jobs....
make[4]: *** [/nix/store/h9xc6fvrqbp3n2wy8m7g4pzyv2r5zz5m-linux-lqx-6.16.7-dev/lib/modules/6.16.7-lqx1/source/scripts/Makefile.build:555: src/fs/bcachefs] Error 2
make[3]: *** [/nix/store/h9xc6fvrqbp3n2wy8m7g4pzyv2r5zz5m-linux-lqx-6.16.7-dev/lib/modules/6.16.7-lqx1/source/Makefile:2001: .] Error 2
make[2]: *** [/nix/store/h9xc6fvrqbp3n2wy8m7g4pzyv2r5zz5m-linux-lqx-6.16.7-dev/lib/modules/6.16.7-lqx1/source/Makefile:248: __sub-make] Error 2
make[2]: Leaving directory &#x27;/build/bcachefs-tools-1.31.2-dkms&#x27;
make[1]: *** [/nix/store/h9xc6fvrqbp3n2wy8m7g4pzyv2r5zz5m-linux-lqx-6.16.7-dev/lib/modules/6.16.7-lqx1/source/Makefile:248: __sub-make] Error 2
make[1]: Leaving directory &#x27;/nix/store/h9xc6fvrqbp3n2wy8m7g4pzyv2r5zz5m-linux-lqx-6.16.7-dev/lib/modules/6.16.7-lqx1/build&#x27;
make: *** [Makefile:8: default] Error 2</pre>
</details>
</details>


---
### `aarch64-linux`
<details>
  <summary>:fast_forward: 2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>bcachefs-tools</li>
    <li>bcachefs-tools.dkms</li>
    <li>linuxPackages_latest.bcachefs (linuxKernel.packages.linux_6_16.bcachefs)</li>
    <li>linuxPackages_latest-libre.bcachefs (linuxKernel.packages.linux_latest_libre.bcachefs)</li>
  </ul>
</details>


---
### `x86_64-darwin`
:white_check_mark: *No rebuilds*


---
### `aarch64-darwin`
:white_check_mark: *No rebuilds*

--- a/nixos/modules/tasks/filesystems/bcachefs.nix
+++ b/nixos/modules/tasks/filesystems/bcachefs.nix
@@ -175,25 +175,6 @@ let
             '';
       };
     };
-
-  assertions = [
-    {
-      assertion =
-        let
-          kernel = config.boot.kernelPackages.kernel;
-        in
-        (
-          kernel.kernelAtLeast "6.7"
-          || (lib.elem (kernel.structuredExtraConfig.BCACHEFS_FS or null) [
-            lib.kernel.module
-            lib.kernel.yes
-            (lib.kernel.option lib.kernel.yes)
-          ])
-        );
-
-      message = "Linux 6.7-rc1 at minimum or a custom linux kernel with bcachefs support is required";
-    }
-  ];
 in
 
 {
@@ -230,7 +211,24 @@ in
   config = lib.mkIf (config.boot.supportedFilesystems.bcachefs or false) (
     lib.mkMerge [
       {
-        inherit assertions;
+        assertions = [
+          {
+            assertion =
+              let
+                kernel = config.boot.kernelPackages.kernel;
+              in
+              (
+                kernel.kernelAtLeast "6.7"
+                || (lib.elem (kernel.structuredExtraConfig.BCACHEFS_FS or null) [
+                  lib.kernel.module
+                  lib.kernel.yes
+                  (lib.kernel.option lib.kernel.yes)
+                ])
+              );
+
+            message = "Linux 6.7-rc1 at minimum or a custom linux kernel with bcachefs support is required";
+          }
+        ];
 
         # Bcachefs upstream recommends using the latest kernel
         boot.kernelPackages = lib.mkDefault pkgs.linuxPackages_latest;
@@ -248,7 +246,6 @@ in
       }
 
       (lib.mkIf ((config.boot.initrd.supportedFilesystems.bcachefs or false) || (bootFs != { })) {
-        inherit assertions;
         boot.initrd.availableKernelModules = [
           "bcachefs"
           "sha256"

--- a/pkgs/by-name/bc/bcachefs-tools/package.nix
+++ b/pkgs/by-name/bc/bcachefs-tools/package.nix
@@ -29,13 +29,13 @@
 
 stdenv.mkDerivation (finalAttrs: {
   pname = "bcachefs-tools";
-  version = "1.31.0";
+  version = "1.31.3";
 
   src = fetchFromGitHub {
     owner = "koverstreet";
     repo = "bcachefs-tools";
     tag = "v${finalAttrs.version}";
-    hash = "sha256-wYlfU4PTcVSPSHbIIDbl8pBOJsBAAl44XBapwFZ528U=";
+    hash = "sha256-sXv6YFM91T08WF5dPU7iQNqWbB/QiL2kMaXm6ZtIDqI=";
   };
 
   nativeBuildInputs = [
@@ -66,7 +66,7 @@ stdenv.mkDerivation (finalAttrs: {
 
   cargoDeps = rustPlatform.fetchCargoVendor {
     src = finalAttrs.src;
-    hash = "sha256-ZCzw3cDpQ8fb2jLYdIWrmlNTPStikIs09jx6jzzC2vM=";
+    hash = "sha256-04YrgYfhZ5NfA2BcF2H6Np1SXRiH6CJpkgc9hzlbMAo=";
   };
 
   makeFlags = [
@@ -100,11 +100,7 @@ stdenv.mkDerivation (finalAttrs: {
   '';
   checkFlags = [ "BCACHEFS_TEST_USE_VALGRIND=no" ];
 
-  postInstall = ''
-    substituteInPlace $out/libexec/bcachefsck_all \
-      --replace-fail "/usr/bin/python3" "${python3.interpreter}"
-  ''
-  + lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform) ''
+  postInstall = lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform) ''
     installShellCompletion --cmd bcachefs \
       --bash <($out/sbin/bcachefs completions bash) \
       --zsh  <($out/sbin/bcachefs completions zsh) \

--- a/pkgs/by-name/bc/bcachefs-tools/package.nix
+++ b/pkgs/by-name/bc/bcachefs-tools/package.nix
@@ -69,16 +69,26 @@ stdenv.mkDerivation (finalAttrs: {
     hash = "sha256-04YrgYfhZ5NfA2BcF2H6Np1SXRiH6CJpkgc9hzlbMAo=";
   };
 
+  outputs = [
+    "out"
+    "dkms"
+  ];
+
   makeFlags = [
     "PREFIX=${placeholder "out"}"
     "VERSION=${finalAttrs.version}"
     "INITRAMFS_DIR=${placeholder "out"}/etc/initramfs-tools"
+    "DKMSDIR=${placeholder "dkms"}"
 
     # Tries to install to the 'systemd-minimal' and 'udev' nix installation paths
     "PKGCONFIG_SERVICEDIR=$(out)/lib/systemd/system"
     "PKGCONFIG_UDEVDIR=$(out)/lib/udev"
   ]
   ++ lib.optional fuseSupport "BCACHEFS_FUSE=1";
+  installFlags = [
+    "install"
+    "install_dkms"
+  ];
 
   env = {
     CARGO_BUILD_TARGET = stdenv.hostPlatform.rust.rustcTargetSpec;

--- a/pkgs/os-specific/linux/bcachefs-kernel-module/default.nix
+++ b/pkgs/os-specific/linux/bcachefs-kernel-module/default.nix
@@ -1,0 +1,48 @@
+{
+  lib,
+  stdenv,
+  bcachefs-tools,
+  kernelModuleMakeFlags,
+  kernel,
+}:
+
+stdenv.mkDerivation {
+  pname = "bcachefs";
+  version = "${kernel.version}-${bcachefs-tools.version}";
+
+  __structuredAttrs = true;
+
+  src = bcachefs-tools.dkms;
+
+  nativeBuildInputs = kernel.moduleBuildDependencies;
+
+  enableParallelBuilding = true;
+
+  makeFlags = kernelModuleMakeFlags ++ [
+    "KDIR=${kernel.dev}/lib/modules/${kernel.modDirVersion}/build"
+    "INSTALL_MOD_PATH=${placeholder "out"}"
+  ];
+
+  installPhase = ''
+    runHook preInstall
+    make -C ${kernel.dev}/lib/modules/${kernel.modDirVersion}/build M=$(pwd) modules_install "''${makeFlags[@]}" "''${installFlags[@]}"
+    runHook postInstall
+  '';
+
+  passthru = {
+    inherit (bcachefs-tools.passthru) tests;
+  };
+
+  meta = {
+    description = "out-of-tree bcachefs kernel module";
+
+    inherit (bcachefs-tools.meta)
+      homepage
+      license
+      maintainers
+      platforms
+      ;
+
+    broken = !(lib.versionAtLeast kernel.version "6.16" && lib.versionOlder kernel.version "6.18");
+  };
+}

--- a/pkgs/top-level/linux-kernels.nix
+++ b/pkgs/top-level/linux-kernels.nix
@@ -375,6 +375,8 @@ in
 
         bbswitch = callPackage ../os-specific/linux/bbswitch { };
 
+        bcachefs = callPackage ../os-specific/linux/bcachefs-kernel-module { };
+
         ch9344 = callPackage ../os-specific/linux/ch9344 { };
 
         chipsec = callPackage ../tools/security/chipsec {

